libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -297,39 +319,14 @@ internal class MapboxTripSession(
      * Start MapboxTripSession
      */
     override fun start(withTripService: Boolean, withReplayEnabled: Boolean) {
-        if (state == TripSessionState.STARTED) {
-            return
-        }
-        navigator.addNavigatorObserver(navigatorObserver)
-        if (withTripService) {
-            tripService.startService()
-        }
-        tripSessionLocationEngine.startLocationUpdates(withReplayEnabled) {
-            updateRawLocation(it)
-        }
-        state = TripSessionState.STARTED
-    }
-
-    private fun updateRawLocation(rawLocation: Location) {
-        val locationHash = rawLocation.hashCode()
-        logD(
-            "updateRawLocation; system elapsed time: ${System.nanoTime()}; " +
-                "location ($locationHash) elapsed time: ${rawLocation.elapsedRealtimeNanos}",
-            LOG_CATEGORY
-        )
-        this.rawLocation = rawLocation
-        locationObservers.forEach { it.onNewRawLocation(rawLocation) }
-        mainJobController.scope.launch(start = CoroutineStart.UNDISPATCHED) {
-            logD(
-                "updateRawLocation; notify navigator for ($locationHash) - start",
-                LOG_CATEGORY
-            )
-            navigator.updateLocation(rawLocation.toFixLocation())
-            logD(
-                "updateRawLocation; notify navigator for ($locationHash) - end",
-                LOG_CATEGORY
-            )
+        if (state != TripSessionState.STARTED) {
+            navigator.addNavigatorObserver(navigatorObserver)
+            if (withTripService) {
+                tripService.startService()
+            }
+            state = TripSessionState.STARTED
         }
+        tripSessionLocationEngine.startLocationUpdates(withReplayEnabled, onRawLocationUpdate)
```

Comment:
Maybe we need to call `NativeNavigaot#resetRideSession` so that native navigator doesn't consider first location updates from new location engine as an error?

Reply:
🤔  yeah. A new location engine seems like a good reason to reset. I'm hesitant, should it go inside the `startTripSession`? Let's consider it in another change. The caller still has the ability to call it or not call it. Right now ReplayRouteSession calls it when the route changes. We can add it there before forcing it everywhere.

https://github.com/mapbox/mapbox-navigation-android/blob/61415ea7ee81831b69fd698b0522bfd69119dd23/libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/route/ReplayRouteSession.kt#L117-L119